File: files/2025-09-14/CSEC 14092025.R
library(tidyverse)
library(readxl)

path = "files/2025-09-14/challenge 60.xlsx"
input = read_excel(path, range = "B2:B18")
test  = read_excel(path, range = "D2:E9")

result = input %>%
  mutate(Stem = Quantity %/% 10, 
         Leaf = Quantity %% 10) %>%
  arrange(Stem, Leaf) %>%
  summarise(Leaf = paste(Leaf, collapse = " "), .by = Stem) 
  
all.equal(result, test)
# [1] TRUE

File: files/2025-09-14/CSEC 14092025.py
import pandas as pd

path = "files/2025-09-14/challenge 60.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=17)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=7)

input['Stem'] = (input['Quantity'] // 10)
input['Leaf'] = (input['Quantity'] % 10)
input = input.sort_values(['Stem', 'Leaf'])
input = input.astype(str)
result = input.groupby('Stem')['Leaf'].apply(' '.join).reset_index()

print(pd.concat([test, result], axis=1))